src/main/java/sirius/db/mixing/DateRange.java

```diff
@@ -29,24 +30,27 @@
 public class DateRange {
 
     private final String key;
-    private final String name;
-    private final LocalDateTime from;
-    private final LocalDateTime until;
+    private final Supplier<String> nameSupplier;
+    private final Supplier<LocalDateTime> fromSupplier;
+    private final Supplier<LocalDateTime> untilSupplier;
 
     /**
-     * Creates a new DateRange with the given unique key, translated (shown) name and two dates specifying the
-     * range
+     * Creates a new DateRange with the given unique key, a supplier that returns the translated (shown) name
+     * and two date suppliers specifying the range
      *
-     * @param key   the unique name of the ranged used as filter value
-     * @param name  the translated name shown to the user
-     * @param from  the lower limit (including) of the range
-     * @param until the upper limit (including) of the range
+     * @param key           the unique name of the ranged used as filter value
+     * @param nameSupplier  the translated name shown to the user
+     * @param fromSupplier  the lower limit (including) of the range
+     * @param untilSupplier the upper limit (including) of the range
      */
-    public DateRange(String key, String name, @Nullable LocalDateTime from, @Nullable LocalDateTime until) {
```

Comment:
add a legacy constructor which invokes a new one and mark this deprecated? this way we prevent a breaking change...